src/miral/basic_window_manager.cpp

```diff
@@ -1503,11 +1503,13 @@ void miral::BasicWindowManager::set_state(miral::WindowInfo& window_info, MirWin
         break;
 
     default:
-        auto const none_active = !active_window();
+        bool const can_become_active =
+            !active_window() ||
+            active_window().application() == window_info.window().application();
```

Comment:
Nit: `window_info.window()` is already cached as `window`

Reply:
So it is